### ISSUE TITLE
unsubscribe button does nothing when clicked

### ISSUE THREAD

**Issue by Alys** [HabitRPG/habitica#3448]
I'm trying to unsubscribe a test account (3b7d59d1-9c79-4c96-a9ac-6e01664d4251), but can't. I click on the gem in HabitRPG's toolbar (production site, not beta), then click on the red "Cancel Subscription" button, and literally nothing happens - no new popup, no change in the existing popup, nothing. This happens on OS X 10.6.8 (Chrome, Firefox, Safari) and Windows 7 (Chrome, Firefox, IE).

In Chrome on OS X, nothing appears in the console when I click the "Cancel Subscription" button, although these messages do appear when the tasks page first loads (i.e, before I click on the gem in the toolbar):

Attr.specified is deprecated. Its value is always true. plusone.js:29
Consider using 'dppx' units instead of 'dpi', as in CSS 'dpi' means dots-per-CSS-inch, not dots-per-physical-inch, so does not correspond to the actual 'dpi' of a screen. In media query expression: (-webkit-min-device-pixel-ratio: 1.5), (min-resolution: 144dpi) tweet_button.1399599786.html:1
Given URL is not allowed by the Application configuration.: One or more of the given URLs is not allowed by the App's settings.  It must match the Website URL or Canvas URL, or the domain must be a subdomain of one of the App's domains. ping?client_id=172525162793917&domain=habitrpg.com&origin=1&redirect_uri=https%3A%2F%2Fs-static.ak.…:1

Tell me if there's anything other information I can provide or any testing I can do.

**Comment by deilann** [comment]
I'm wondering if the subscription is canceled, but because of the change to
keep you subscribed until the end of the month, it's not visibly showing
anything to denote that.

So you're subscribed -- but not subscribed.

SUBSCRIPTION LIMBO

**Comment by lefnire** [comment]
On the subscription page, it should show you that you have an active subscription which will be terminated a month from your last payment ([screenshot](http://gyazo.com/0086ecab8bb3c1620df638a83bdae653)).

**Comment by Alys** [comment]
No, I'm afraid not; there's no indication that clicking on the "Cancel Subscription" button had any effect. Screenshot below, and the json data is:

"plan":{"customerId":"cus_xxxxxxxxxxxxxx","dateCreated":"2014-04-13T06:58:18.017Z","dateUpdated":"2014-05-01T19:43:13.762Z","paymentMethod":"Stripe","planId":"basic_earned","mysteryItems":[],"gemsBought":25},

![screen shot 2014-05-12 at 7 11 12 am](https://cloud.githubusercontent.com/assets/1495809/2939609/30fc1f3a-d951-11e3-8aea-ba1f0cec11d1.png)

**Comment by lefnire** [comment]
Ah, the button's literally doing nothing for you then (since the database entry shows no cancellation effects). Hmm, strange there's no javascript errors. I've seen this happen to someone before as well, and I was able to use their account to cancel the sub just fine... *scratches head* let me brew on this

**Comment by Alys** [comment]
Yup, literally nothing happens and I mean "literally" literally. :) I would be happy to give you the account's password if you'd like that for testing (obviously you have full visibility of the account's contents anyway, but the password would let you mimic what a non-admin user would do). Shall I email it to admin@habitrpg.com? There is nothing at all personal in the account so I don't really care if the password is intercepted.

**Comment by lefnire** [comment]
This is gonna scare you... but no need, as long as I have your permission :)

**Comment by lefnire** [comment]
Just so it doesn't scare too much, I can use a manual UUID + API Token login (I don't have access to passwords).

**Comment by Alys** [comment]
Ha, you will have to try harder to scare me. :) I would have been
astonished if you did have access to passwords because you're clearly smart
enough to ensure they are encrypted properly. I'm not surprised that you
have a way to log in to the website with the token; that just makes sense
for testing. You certainly have my permission.

**Comment by lefnire** [comment]
yeesh, yet again - I was able to click the button without a hitch on my end. Maybe an Angular `ng-click` compatibility for certain users' browsers? I'm not sure how to test this one..

**Comment by Alys** [comment]
I'm getting the "will become inactive" message now.

Did you see the list of browser that I tested on? Six in total, spread over
two different machines. It is certainly odd! If you do think of any tests
that might help, be sure to tell me. I won't mind resubscribing and trying
many different things if you want to clutch at straws. :)

**Comment by lefnire** [comment]
It's definitely important, I just don't have a clue how to go about this. We get up to 10 unsubscriptions per day. I've only ever seen someone (besides yourself) report this issue once, so it's very rare. And without JS issues (as you mentioned), I'm blank.

I'll do some more testing when I get a moment. If you have a local install, running in development mode will set credit-card payments to [accept dummy credit cards](https://stripe.com/docs/checkout).

**Comment by Alys** [comment]
HA! I've worked it out. If I click on the gear icon ("Settings") then on
"Subscription", I get to this page:
https://habitrpg.com/#/options/settings/subscription
and then clicking on "Cancel Subscription" pops up a confirmation alert box
(I assume that's what you do whenever you test unsubscribing). However I
don't get the alert box when I access the  "Cancel Subscription" button as
described in my OP for this issue. The
confirm(window.env.t('sureCancelSub') code doesn't work when called from
within the Gems screen.

EDIT: I've documented this on the wiki: http://habitrpg.wikia.com/wiki/Subscription#Unsubscription_Procedure
